arrow/src/compute/kernels/cast.rs

```
@@ -350,6 +351,32 @@ macro_rules! cast_decimal_to_integer {
     }};
 }
 
+// cast the string array to defined decimal data type array
+macro_rules! cast_string_to_decimal {
+    ($ARRAY: expr, $ARRAY_TYPE: ident, $PRECISION : ident, $SCALE : ident) => {{
+        let mul = 10_f64.powi(*$SCALE as i32);
+        let decimal_array = $ARRAY
+            .as_any()
+            .downcast_ref::<$ARRAY_TYPE>()
+            .unwrap()
+            .iter()
+            .map(|val| match val {
+                Some(val) => {
+                    let val: Option<f64> = lexical_core::parse(val.as_bytes()).ok();
```

This should throw an error + map_error -> ArrowError